src/components/CmsZone.vue

```diff
@@ -194,36 +196,94 @@ class ZoneObserverManager {
 
 @Component({
   name: 'cms-zone',
-  components: { CmsCarousel, CmsContent, CmsInspectSheet },
+  components: { CmsCarousel, CmsContent, CmsInspectSheet, CmsIntersectionObserver },
 })
 export default class CmsZone extends Vue {
   @Prop(String) zoneId!: string;
   @Prop(Object) extra!: {};
   @Prop(Object) context!: {};
+  @Prop(Object) intersectionOptions!: {};
 
   zoneStatus: string | null = null;
-  zoneType: string = '';
-  zoneHeader: string = '';
-  zoneFooter: string = '';
+  lastResponse: CMSZoneResponse | null = null;
   contents: Content[] = [];
-  cursor: string = '';
-  scrollable: Element | null = null;
+  observed: Element[] = [];
+  shouldShowInspectModal = false;
 
   nonce: number = 0;
   cursorLoading: boolean = false;
-  next = debounce(() => this.getNextPage(), 400);
-  observed: Element[] = [];
+  haltPaging: boolean = false;
 
-  shouldShowInspectModal = false;
+  async startPaging() {
+    if (!this.cursorLoading) {
+      this.fetchPages();
+    }
+  }
+
+  async fetchPages() {
+    this.cursorLoading = true;
+
+    try {
+      while (!this.haltPaging && !this.allContentLoaded) {
+        await this.getNextPage();
+        await Vue.nextTick();
```

Comment:
If i'm reading this right, `this.getNextPage()` already has a nextTick. Do we need this one?

Reply:
Probably not, will try and remove.